drivers/pinctrl/pinctrl_iopctl.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_iopctl
+
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/init.h>
+
+#define OFFSET(mux) (((mux) & 0xFFF00000) >> 20)
+#define INDEX(mux)  (((mux) & 0xF0000) >> 16)
+
+/* IOPCTL register addresses. */
+static uint32_t *iopctl[] = {
+#if (DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(iopctl0)))
+	(uint32_t *)DT_REG_ADDR(DT_NODELABEL(iopctl0)),
+#else
+	NULL,
+#endif
+#if (DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(iopctl1)))
+	(uint32_t *)DT_REG_ADDR(DT_NODELABEL(iopctl1)),
+#else
+	NULL,
+#endif
+#if (DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(iopctl2)))
+	(uint32_t *)DT_REG_ADDR(DT_NODELABEL(iopctl2)),
+#else
+	NULL,
+#endif
+};
+
+int pinctrl_configure_pins(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt, uintptr_t reg)
+{
+	for (uint8_t i = 0; i < pin_cnt; i++) {
+		uint32_t pin_mux = pins[i];
```

Comment:
Do we need to have a mask here as well? It seems like we should not be writing all the bits of the pinmux setting, since the upper bits are used for the pin/offset

Reply:
Yes, I have added code to use masked value to write into iopctl register. Thank you.